### DISCUSSION TITLE
HMR not working in authored module

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [nuxt/nuxt#24385] [Questions] [answered]
Hi, I've had this issue many times before in many nuxt versions, but it was today that I gave it a deep thought.

HMR is not working inside files in `~/modules`. I made a `~/modules/chat.ts` module today to initialize a SocketIO server,
and every time I make a change in that file, there's no HMR, so I have to stop nuxt dev server and restart.
I change any other file like a page, a server api, a plugin, a middleware, and nuxt.config.ts, and HMR works immediately.

Is this by design or a bug? Thanks. 💚🔥

**Comment by akasection** [comment, marked as answer]
IIRC HMR only works on application source code, while modules more likely refer to the builders. But if your modules are installing plugins, components, utils or pages, I think you can still register HMR to those files via `import.meta.hot`. The module entry file still need you to restart the nuxt server.

This PR is an example of transforming a file (app.config.ts) into HMR-ready: https://github.com/nuxt/framework/pull/6333/files#diff-421426fc478e91dfcb94a9b4a10fa8c38e6c30a0546606fe7a6baaf05837adf8R35
